pallets/dkg-metadata/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+// Copyright (C) 2020-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::Encode;
+
+use frame_support::{traits::OneSessionHandler, Parameter};
+
+use core::convert::TryFrom;
+use sp_runtime::{
+	generic::DigestItem,
+	traits::{IsMember, Member},
+	RuntimeAppPublic,
+};
+use sp_std::prelude::*;
+
+use dkg_runtime_primitives::{
+	traits::OnAuthoritySetChangeHandler, AuthorityIndex, AuthoritySet, ConsensusLog, DKG_ENGINE_ID,
+};
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Authority identifier type
+		type DKGId: Member + Parameter + RuntimeAppPublic + Default + MaybeSerializeDeserialize;
+
+		/// Listener for authority set changes
+		type OnAuthoritySetChangeHandler: OnAuthoritySetChangeHandler<
```

Comment:
Thinking if we want to plug multiple pallets to receive changes to the authority set, we could make this a Tuple, but this works for now